### PR TITLE
tls: more generic function to set cafile and capath

### PR DESCRIPTION
**Pull request by cHuberCoffee**

### CODE REVIEW

File: src/tls/openssl/tls.c [hunk 1]
```diff
@@ -249,12 +249,32 @@ int tls_alloc(struct tls **tlsp, enum tls_method method, const char *keyfile,
...
-	if (!(SSL_CTX_load_verify_locations(tls->ctx, cafile, NULL))) {
-		DEBUG_WARNING("Can't read CA file: %s\n", cafile);
+	if (!(SSL_CTX_load_verify_locations(tls->ctx, cafile, capath))) {
+		if (str_isset(cafile))
+			DEBUG_WARNING("Can't read CA file: %s\n", cafile);
+		if (str_isset(capath))
+			DEBUG_WARNING("Can't read CA path: %s\n", capath);
+
 		ERR_clear_error();
 		return EINVAL;
```
Comment (sreimers) [inline src/tls/openssl/tls.c:279]: Maybe `ENOENT` is a better return code.

File: src/tls/openssl/tls.c [hunk 2]
```diff
@@ -249,12 +249,32 @@ int tls_alloc(struct tls **tlsp, enum tls_method method, const char *keyfile,
...
 
 	/* Load the CAs we trust */
-	if (!(SSL_CTX_load_verify_locations(tls->ctx, cafile, NULL))) {
-		DEBUG_WARNING("Can't read CA file: %s\n", cafile);
+	if (!(SSL_CTX_load_verify_locations(tls->ctx, cafile, capath))) {
+		if (str_isset(cafile))
+			DEBUG_WARNING("Can't read CA file: %s\n", cafile);
+		if (str_isset(capath))
+			DEBUG_WARNING("Can't read CA path: %s\n", capath);
+
```
Comment (sreimers) [inline src/tls/openssl/tls.c:277]: Can you add a negative test case (check for wrong cafile/capath) for `tls_add_cafile_path` in https://github.com/baresip/retest?

**Review by sreimers — CHANGES_REQUESTED** [review]

File: src/tls/openssl/tls.c [hunk 3]
```diff
@@ -249,14 +249,37 @@ int tls_alloc(struct tls **tlsp, enum tls_method method, const char *keyfile,
...
+ */
+int tls_add_cafile_path(struct tls *tls, const char *cafile,
+	const char *capath)
+{
+	if (!tls || (!cafile && !capath) || !tls->ctx)
 		return EINVAL;
 
+	if (capath && !fs_isdir(capath)) {
+		DEBUG_WARNING("capath is not a directory\n");
+		return ENOENT;
```
Comment (sreimers) [inline src/tls/openssl/tls.c:273]: There is a `ENOTDIR` for this case.

**Review by sreimers — APPROVED** [review]

**Comment by juha-h** [comment]
These should be improved:
```
 * @param cafile PEM file with CA certificate
 * @param capath Path containing CA certificates
```
Isn't it so that also cafile file can contain many ca certificates?

Path does not contain ca certificates, but files than contain ca certificates.

**Pull request merged**